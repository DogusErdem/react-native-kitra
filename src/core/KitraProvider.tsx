import type { ComponentType, ForwardedRef } from 'react';
import React from 'react';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import Animated from 'react-native-reanimated';
import { StyleProp, ViewStyle } from 'react-native';

import { ThemeProvider } from './theme/theme';
import { TypographyContext, TypographyProvider } from './typography/typography';
import { NotificationContextType, NotificationProvider } from './notification/notification';

const Feather = require('react-native-vector-icons/Feather').default;
const Ionicons = require('react-native-vector-icons/Ionicons').default;
const AntDesign = require('react-native-vector-icons/AntDesign').default;

Animated.addWhitelistedNativeProps({ text: true });
const messageTypes = (theme:any) => ({
  SUCCESS: {
    backgroundColor: theme.colors.status.successLight,
    icon: <Feather name="check" size={25} color={theme.colors.status.success} />,
  },
  INFO: {
    backgroundColor: theme.colors.system.primary15,
    icon: <Ionicons name="information-circle-outline" size={25} color={theme.colors.system.primary} />,
  },
  DANGER: {
    backgroundColor: theme.colors.status.errorLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.error} />,
  },
  WARNING: {
    backgroundColor: theme.colors.status.warningLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.warning} />,
  },
});

type KitraProviderType= {
  children?:any,
  limit?:number,
  notificationRef?:React.Ref<NotificationContextType>,
  messageType?:(theme?:any, colorScheme?:'dark' | 'light') => {[key:string]: {
      backgroundColor:string,
    icon?:React.ReactNode
    onPress?:()=>void
  }},
  notificationCcontainerStyle?:StyleProp<ViewStyle>,
  customView?:({ header, type, message, theme }:
    {header:string, type:string, message:string, theme:any})=>React.ReactNode}

export const KitraProvider =
({ children, messageType = messageTypes, notificationCcontainerStyle, limit, customView,
  notificationRef }: KitraProviderType) => (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <ThemeProvider>
        <TypographyProvider>
          <NotificationProvider
            messageType={theme => (messageType ? messageType(theme) : theme)}
            notificationCcontainerStyle={notificationCcontainerStyle}
            limit={limit}
            customView={item => customView?.(item)}
            ref={notificationRef}
          >
            {children}
          </NotificationProvider>
        </TypographyProvider>
      </ThemeProvider>
    </GestureHandlerRootView>
);

export function applyDefaults<T>(Component: ComponentType<T>) {
  return React.forwardRef((props: T, ref: ForwardedRef<any>) => (
    <TypographyContext.Consumer>
      {typography =>
        <Component ref={ref} typography={typography?.typography} {...props} />
          }
    </TypographyContext.Consumer>
  ));
}
