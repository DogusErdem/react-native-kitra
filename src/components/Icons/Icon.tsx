import { getIconType } from '../../utilities';
import { applyDefaults } from '../../core/KitraProvider';
import type { FCCWD, IconComponentProps, IconType } from '../../types';

type IconProps = {
    type: IconType,
} & IconComponentProps;

const Icon:FCCWD<IconProps> = ({ type, ...props }) => {
  const IconComponent = getIconType(type || 'ionicons');

  return (
    <IconComponent {...props} />
  );
};

export default applyDefaults(Icon);
