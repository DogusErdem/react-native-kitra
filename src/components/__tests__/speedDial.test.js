import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react-native';
/* eslint-disable no-undef */
import SpeedDial from '../SpeedDial/SpeedDial';
import Icon from '../Icons/Icon';
import { KitraProvider } from '../../core/KitraProvider';
// Mocked functions for onPress
const onPressItem1 = jest.fn();
const onPressItem2 = jest.fn();
const onPressItem3 = jest.fn();
const renderWithContext = children => render(<KitraProvider>{children}</KitraProvider>);

const items = [
  { icon: <Icon type="ant-design" name="check" size={25} color="#ffffff" />, onPress: onPressItem1 },
  { title: 'ads', onPress: onPressItem2 },
  { icon: <Icon type="ant-design" name="check" size={25} color="#ffffff" />, onPress: onPressItem3 },
];
describe('SpeedDial', () => {
  describe('renders speedDial with variant prop', () => {
    it('variant is spread', () => {
      const tree = renderWithContext(<SpeedDial
        items={items}
        variant="spread"
        baseItemIcon={<Icon type="material-community" name="close" size={25} color="white" />}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('variant is flat', () => {
      const tree = renderWithContext(<SpeedDial
        items={items}
        variant="flat"
        baseItemIcon={<Icon type="ant-design" name="check" size={20} />}
      />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('renders speedDial elemts onpress', () => {
    it('onpress baseitem', () => {
      renderWithContext(<SpeedDial items={items} />).toJSON();
      const test = screen.getByTestId('base_item');
      fireEvent.press(test);
    });
    it('onpress childitem', () => {
      renderWithContext(<SpeedDial
        items={items}
        variant="flat"
        baseItemIcon={<Icon type="ant-design" name="check" size={20} />}
      />).toJSON();
      const test = screen.getByTestId('base_item');
      fireEvent.press(test);

      const child1 = screen.getByTestId('children_item_0');
      fireEvent.press(child1);
      expect(onPressItem1).toHaveBeenCalled();

      const child2 = screen.getByTestId('children_item_1');
      fireEvent.press(child2);
      expect(onPressItem2).toHaveBeenCalled();

      const child3 = screen.getByTestId('children_item_2');
      fireEvent.press(child3);
      expect(onPressItem3).toHaveBeenCalled();
    });
  });
});
