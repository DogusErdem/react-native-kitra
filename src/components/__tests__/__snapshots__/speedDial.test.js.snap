// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedDial renders speedDial with variant prop variant is flat 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "bottom": 30,
          "position": "absolute",
          "right": 30,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            [
              {
                "opacity": 0,
                "transform": [
                  {
                    "translateY": -0,
                  },
                ],
              },
              {
                "bottom": 20,
              },
            ],
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_0"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#ffffff",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            î˜²
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            [
              {
                "opacity": 0,
                "transform": [
                  {
                    "translateY": -0,
                  },
                ],
              },
              {
                "bottom": 20,
              },
            ],
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "fontSize": 12,
                "fontWeight": "500",
                "paddingHorizontal": 12,
                "paddingVertical": 5,
                "position": "absolute",
                "right": 60,
                "textAlign": "center",
                "top": 10,
              },
              {
                "backgroundColor": "white",
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#82B98E",
                  "fontSize": 10,
                  "lineHeight": 16,
                },
              ]
            }
          >
            ads
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_1"
        />
      </View>
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            [
              {
                "opacity": 0,
                "transform": [
                  {
                    "translateY": -0,
                  },
                ],
              },
              {
                "bottom": 20,
              },
            ],
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_2"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#ffffff",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            î˜²
          </Text>
        </View>
      </View>
      <View
        style={
          [
            null,
            {
              "borderRadius": 50,
              "height": 60,
              "width": 60,
            },
            {
              "backgroundColor": "#195CEF",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="base_item"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            î˜²
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`SpeedDial renders speedDial with variant prop variant is spread 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "bottom": 30,
          "position": "absolute",
          "right": 30,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            {
              "opacity": 0,
              "transform": [
                {
                  "translateX": -0,
                },
              ],
            },
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_0"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#ffffff",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            î˜²
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            {
              "opacity": 0,
              "transform": [
                {
                  "translateX": -0,
                },
                {
                  "translateY": -0,
                },
              ],
            },
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_1"
        />
      </View>
      <View
        style={
          [
            {
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "position": "absolute",
              "width": 50,
            },
            {
              "opacity": 0,
              "transform": [
                {
                  "translateY": -0,
                },
              ],
            },
            {
              "backgroundColor": "#82B98E",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="children_item_2"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#ffffff",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            î˜²
          </Text>
        </View>
      </View>
      <View
        style={
          [
            null,
            {
              "borderRadius": 50,
              "height": 60,
              "width": 60,
            },
            {
              "backgroundColor": "#195CEF",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 50,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="base_item"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "white",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            typography={
              {
                "body": {
                  "medium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  "regular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "lineHeight": 16,
                  },
                  "smedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 12,
                    "lineHeight": 18,
                  },
                  "sregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                  "xsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 10,
                    "lineHeight": 16,
                  },
                  "xsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                    "lineHeight": 14,
                  },
                  "xxsmedium": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                  "xxsregular": {
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 8,
                    "lineHeight": 12,
                  },
                },
                "heading": {
                  "h1": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 30,
                    "lineHeight": 45,
                  },
                  "h2": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 28,
                    "lineHeight": 42,
                  },
                  "h3": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  "h4": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 19,
                    "lineHeight": 29,
                  },
                  "h5": {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 16,
                    "lineHeight": 25,
                  },
                },
              }
            }
          >
            ó°…–
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
